pytorch_lightning/trainer/trainer.py

```diff
@@ -172,7 +171,7 @@ def __init__(
         weights_save_path: Optional[str] = None,  # TODO: Remove in 1.8
         num_sanity_val_steps: int = 2,
         resume_from_checkpoint: Optional[Union[Path, str]] = None,
-        profiler: Optional[Union[BaseProfiler, str]] = None,
+        profiler: Optional[Union[Profiler, str]] = None,
```

Comment:
@carmocca Should I undo the change here for someone to address it in a different PR (#13046)? This PR needs it fixed for merge.

Reply:
If this PR gets merged before it is fixed in the other PR, we can still apply the fix to the bugfix branch directly